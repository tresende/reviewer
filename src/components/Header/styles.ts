import styled, { css } from 'styled-components'
import media from 'styled-media-query'

const breakpoint = 'medium'

export const Nav = styled.nav`
  max-width: 1100px;
  margin: 0 auto;
`

export const Icon = styled.label`
  ${({ theme }) => css`
    display: none;
    background-color: ${theme.colors.primary};
    margin: ${theme.spacings.xxsmall};
    padding: ${theme.spacings.xxsmall};
    border-radius: ${theme.border.smallRadius};
  `}

  ${media.lessThan(breakpoint)`
    display: block;
    color: white;
    font-style: normal;
  `}
`

export const Items = styled.div`
  display: flex;
  align-items: center;
  ${media.lessThan(breakpoint)`
    flex-direction: column;
  `}
`

export const MainMenu = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex: 1;
    justify-content: space-around;
    align-items: center;
    color: ${theme.colors.white};
    font-weight: ${theme.font.bold};
  `}
  ${media.lessThan(breakpoint)`
    flex-direction: column;
    width:100%;
    padding: 0 1rem;
  `}
`

export const UserMenu = styled(MainMenu)`
  display: flex;
  justify-content: flex-end;
`

export const MenuItem = styled.div`
  ${({ theme }) => css`
    text-align: center;
    font-family: 'Montserrat';
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.white};
    ${media.lessThan(breakpoint)`
      padding: ${theme.spacings.xxsmall} 0;
      border-bottom: ${theme.border.default} ${theme.colors.gray500};
      flex-direction: column;
      width: 100%;
      padding: 1rem;
      display:none;
    `}
  `}
`

export const Logo = styled(MenuItem)`
  ${media.lessThan(breakpoint)`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding:0;
    border:0;
  `}
`

type HeaderButtonProps = {
  color?: 'primary' | 'white'
}

export const HeaderButton = styled.button<HeaderButtonProps>`
  ${({ theme, color = 'primary' }) => css`
    border: 2px solid ${theme.colors[color]};
    border-radius: 30px;
    background-color: transparent;
    color: ${theme.colors[color]};
    font-weight: ${theme.font.bold};
    padding: ${theme.spacings.xsmall};
    ${media.lessThan(breakpoint)`
      width:100%;
      margin-top:1rem;
      justify-content: center;
      padding:  ${theme.spacings.xxsmall};
      display:none;
    `}
  `}
`

export const Subitem = styled.span`
  ${({ theme }) => css`
    margin: ${theme.spacings.xxsmall};
    font-size: ${theme.font.sizes.small};
    display: none;
  `}
`

const WrapperModifiers = {
  open: () => css`
    ${MenuItem},
    ${HeaderButton},
    ${Subitem} {
      display: flex;
    }
  `,
  close: () => css`
    ${Logo} {
      display: flex;
    }
  `
}

export const Wrapper = styled.header<{ open: boolean }>`
  ${({ theme, open }) => css`
    background-color: ${theme.colors.black};
    padding: 1rem;
    input {
      display: none;
    }

    ${open && WrapperModifiers.open()};
    ${!open && WrapperModifiers.close()};
  `}
`
