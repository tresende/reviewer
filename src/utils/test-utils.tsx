import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
