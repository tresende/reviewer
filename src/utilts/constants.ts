const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MAX_RATING = 5
export { SCORE_LABEL, MAX_RATING }
