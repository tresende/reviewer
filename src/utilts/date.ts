export const toPTBRFormat = (date: Date | string) => {
  let toFormat = date
  if (typeof date === 'string') toFormat = new Date(date)
  const formatedDate = new Intl.DateTimeFormat('pt-br').format(toFormat)
  return formatedDate
}
