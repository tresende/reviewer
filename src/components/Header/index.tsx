import * as S from './styles'

const Header = () => (
  <S.Wrapper>
    <S.Icon htmlFor="hamburger">&#9776;</S.Icon>
    <S.Toggle type="checkbox" id="hamburger" />
    <S.Items id="items">
      <S.MenuItem>
        <img src="img/logo.png" alt="Beer Or Coffee" />
      </S.MenuItem>
      <S.MenuItem>Espa√ßos de Coworking</S.MenuItem>
      <S.MenuItem>Planos</S.MenuItem>
      <S.MenuItem>Descontos</S.MenuItem>
      <S.MenuItem style={{ flex: 3 }}>
        <S.HeaderButton>Adiquirir um Plano</S.HeaderButton>
        <S.Subitem>Empresas</S.Subitem>
        <S.HeaderButton color="white">Oi, Thiago</S.HeaderButton>
      </S.MenuItem>
    </S.Items>
  </S.Wrapper>
)

export default Header
