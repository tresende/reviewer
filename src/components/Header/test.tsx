import { render } from 'utilts/test-utils'
import Header from '.'

describe('<Header />', () => {
  it('should render the header', () => {
    const { container } = render(<Header />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
