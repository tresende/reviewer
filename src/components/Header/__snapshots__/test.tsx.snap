// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the header 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #222222;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-image: url('/img/logo.png');
  background-repeat: no-repeat;
  background-position: 8rem 1rem;
  padding: 0 1.6rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 20;
  padding: 0 14rem;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  font-family: 'Montserrat';
  font-size: 1.4rem;
  font-weight: 700;
  background-color: #222222;
  z-index: 20;
  padding: 0.8rem 0;
  color: #FFFFFF;
  cursor: pointer;
}

.c2 {
  display: none;
}

.c1 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  display: none;
  background-color: #FABA40;
  margin: 0.8rem;
  padding: 0.8rem;
  border-radius: 0.4rem;
}

.c5 {
  border: 2px solid #FABA40;
  color: #FABA40;
  background-color: transparent;
  padding: 1.6rem;
  border-radius: 30px;
  font-weight: 700;
}

.c7 {
  border: 2px solid #FFFFFF;
  color: #FFFFFF;
  background-color: transparent;
  padding: 1.6rem;
  border-radius: 30px;
  font-weight: 700;
}

.c6 {
  margin: 1rem 0;
}

@media (max-width:1170px) {
  .c0 {
    background-position: 1rem 1rem;
    padding: 1rem 1.6rem;
  }

  .c0 input:checked ~ #items {
    display: block;
  }
}

@media (max-width:1170px) {
  .c3 {
    display: none;
    padding: 0;
    text-align: left;
  }
}

@media (max-width:1170px) {
  .c4 {
    border-bottom: 1px solid #4b4b4b;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:1170px) {
  .c1 {
    display: block;
    color: white;
    font-style: normal;
  }
}

@media (max-width:1170px) {
  .c5 {
    width: 100%;
    padding: 0.8rem;
  }
}

@media (max-width:1170px) {
  .c7 {
    width: 100%;
    padding: 0.8rem;
  }
}

<nav
  class="c0"
>
  <label
    class="c1"
    for="hamburger"
  >
    ☰
  </label>
  <input
    class="c2"
    id="hamburger"
    type="checkbox"
  />
  <div
    class="c3"
    id="items"
  >
    <div
      class="c4"
    >
      Espaços de Coworking
    </div>
    <div
      class="c4"
    >
      Planos
    </div>
    <div
      class="c4"
    >
      Descontos
    </div>
    <div
      class="c4"
      style="flex: 3;"
    >
      <button
        class="c5"
      >
        Adiquirir um Plano
      </button>
      <span
        class="c6"
      >
        Empresas
      </span>
      <button
        class="c7"
        color="white"
      >
        Oi, Thiago
      </button>
    </div>
  </div>
</nav>
`;
